test/bdd/pkg/introduce/introduce_sdk_steps.go

```diff
@@ -396,3 +407,43 @@ func (a *SDKSteps) sendRequest(introducee1, introducer, introducee2 string) erro
 
 	return a.clients[introducee1].SendRequest(to, conn1.MyDID, conn1.TheirDID)
 }
+
+func (a *SDKSteps) newOOBRequest(agentID string) (*outofband.Request, error) {
+	client, err := outofband.New(a.bddContext.AgentCtx[agentID])
+	if err != nil {
+		return nil, err
+	}
+
+	r, err := client.CreateRequest(
+		outofband.WithLabel(agentID),
+		outofband.WithAttachments(&decorator.Attachment{
+			ID:          uuid.New().String(),
+			Description: "test",
+			Data: decorator.AttachmentData{
+				JSON: map[string]interface{}{},
+			},
+		},
+		))
+	if err != nil {
+		return nil, err
+	}
+
+	return r, nil
+}
+
+//  creates clients for other protocols (eg. out-of-band, did-exchange)
+func (a *SDKSteps) createExternalClients(participants string) error {
+	if err := a.didExchangeSDKS.CreateDIDExchangeClient(participants); err != nil {
+		return err
+	}
+
+	if err := a.outofbandSDKS.CreateClients(participants); err != nil {
+		return err
+	}
+
+	if err := a.didExchangeSDKS.RegisterPostMsgEvent(participants, "completed"); err != nil {
+		return err
+	}
+
+	return nil
+}
```

Comment:
The last error check in the function usually is redundant. I prefer the following:
`return a.didExchangeSDKS.RegisterPostMsgEvent(participants, "completed");`

Reply:
See my previous comment.